internal/sys/fs.go

```diff
@@ -61,21 +61,72 @@ func (w *stdioFileWriter) Close() error {
 	return nil
 }
 
-type stdioFileReader struct {
-	r io.Reader
-	s fs.FileInfo
+// StdioFilePoller is a strategy for polling a StdioFileReader for a given duration.
```

Comment:
The changes to this file have expanded significantly, maybe it's worth moving `Poller*` to a separate file?

Reply:
I am also starting to wonder whether this change really is beneficial or if all I am doing is testing my own mocks... :/ 

The basic issue is that I wanted to be able to simulate various stdin scenarios and I could not find any other way other than mocking (I thought `os.Pipe()` would return no data with `Select()` but even those immediately return "data available" even when empty)

if I revert all these "poller" changes and restore `fs` then we lose the ability to simulate all these scenarios in tests (e.g. "a stdin that's not written for N millis"), but on the other hand, we are not really testing `select` but the behavior of the Poller (?)